_posts/2021-04-08-alekos-bdk.md

```diff
@@ -0,0 +1,100 @@
+support a very limited set of operations, or they would trade some of the
+convenience for a more powerful API, making them especially tricky to use in
+practice without a deep understanding of the protocol.
+
+This ultimately has reduced the “diversity” of Bitcoin
+products and wallets available to the public today: the vast majority of them
+only support single-signature schemes, or at most simple multi-signatures. More
+complex scripts that involve, for instance, timelocks are very rarely seen in
+the wild.
+
+The advent of output descriptors promises to remove once and for all the
+trade-off between usability and power: descriptors can be used to express with a
+very concise syntax even the most complex Bitcoin spending conditions. With the
+right tools, a developer can build a product that relies on arbitrary spending
+policies without having to ever touch raw Bitcoin scripts or transactions.
+
+Even better, software built with descriptors in mind can be very easy to adapt
+to new protocol changes, like the upcoming Taproot soft-fork, and it allows
+prototyping and shipping new features much more quickly because most of the
+complexity is handled internally by the library. Developers that adopt this
+technology can ultimately spend more time working on their product and tailoring
+it to their users’ needs, rather than having to learn all the low level details
+of the Bitcoin protocol and having to keep up with them.
+
+The [Bitcoin Dev Kit][bdk website] (or BDK for short) project is working on building a set of
+open source libraries that can be used to develop descriptor-based Bitcoin
+products. Its core components are written in Rust, a language that prioritizes
+memory-safety while at the same time being “low-level” enough that it can be
+easily compiled and run on embedded hardware. BDK is also very well integrated
+with the larger rust-bitcoin ecosystem and all the major libraries can easily
+interoperate with each other.
+
+BDK tries to be very modular and provides a few ready-to-use default modules out
+of the box: one can use the provided branch-and-bound coin selection, or swap it
+with a custom algorithm without having to fork and modify the library. The same
+applies to the internal database and the blockchain backend: the library
+provides the implementation for different databases and backend clients based on
+Electrum and Neutrino. Those are bundled with the library, but they can be
+replaced (or removed entirely to make air-gapped ephemeral wallets) to better
+suit the developers’ needs.
+
+This allows BDK to be potentially used in many different kinds of products that
+need to interact with the Bitcoin network: it can be the watch-only wallet of a
+vending machine running on a low-power board, the high security corporate
+Bitcoin vault that runs on multiple air-gapped geographically-distributed
+computers, or everything else in between.
+
+I started this project more than a year ago, and I have been leading its
```

Comment:
I find the shift from third-person description to first-person narrative in the middle of the document to be a bit disconcerting.  I suggest you look for a way to insert yourself earlier in the document, preferably by the second paragraph.